sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusSessionProcessor.cs

```diff
@@ -30,10 +30,9 @@ public class ServiceBusSessionProcessor : IAsyncDisposable
         public virtual string EntityPath => InnerProcessor.EntityPath;
 
         /// <summary>
-        /// Gets the ID to identify this processor. This can be used to correlate logs and exceptions.
+        /// Gets the Identifier used to identify this processor client.  If <c>null</c> or empty, a random unique value will be will be used.
```

Comment:
Seems like we only generate the random ID if it is null, not if it is empty. I'd be okay with updating the behavior to match the docs or vice versa.

Reply:
Event Hubs is testing null or empty:   _([src](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs#L245))_

```csharp
Identifier = string.IsNullOrEmpty(clientOptions.Identifier)
    ? Guid.NewGuid().ToString()
    : clientOptions.Identifier;
```